packages/dds/merge-tree/src/localReference.ts

```diff
@@ -41,6 +41,8 @@ import { minReferencePosition,
      */
     public segment: ISegment | undefined;
 
+    public pending: boolean = false;
```

Comment:
Can interval store it as a property if it needs it, rather than pushing to local reference

Reply:
This will be needed by the code in markRangeRemoved to not slide references whose creation has not been acked.

Reply:
I'm worried that we are losing separation of concerns here. ideally the merge tree really only knows about local references but does sliding in a way to support structures like interval collections. 

I wonder if there are alternative here where the behavior more naturally falls out. for instance, what if pending creates had a different ref type (simple) and on ack they are converted to slide on remove?